Sources/Testing/Expectations/Expectation+Macro.swift

```diff
@@ -572,3 +572,37 @@ public macro require(
   sourceLocation: SourceLocation = #_sourceLocation,
   performing expression: @escaping @Sendable @convention(thin) () async throws -> Void
 ) -> ExitTest.Result = #externalMacro(module: "TestingMacros", type: "ExitTestRequireMacro")
+
+/// Capture a sendable and codable value to pass to an exit test.
+///
+/// - Parameters:
+///   - value: The captured value.
+///   - name: The name of the capture list item corresponding to `value`.
+///
+/// - Returns: `value` verbatim.
+///
+/// - Warning: This macro is used to implement the `#expect(processExitsWith:)`
+///   macro. Do not use it directly.
+@freestanding(expression)
+public macro __capturedValue<T>(
+  _ value: T,
+  _ name: String
+) -> T = #externalMacro(module: "TestingMacros", type: "ExitTestCapturedValueMacro") where T: Sendable & Codable
+
+/// Emit a compile-time diagnostic when an unsupported value is captured by an
+/// exit test.
+///
+/// - Parameters:
+///   - value: The captured value.
+///   - name: The name of the capture list item corresponding to `value`.
+///
+/// - Returns: The result of a call to `fatalError()`. `value` is discarded at
+///   compile time.
+///
+/// - Warning: This macro is used to implement the `#expect(processExitsWith:)`
+///   macro. Do not use it directly.
+@freestanding(expression)
+public macro __capturedValue<T>(
+  _ value: borrowing T,
+  _ name: String
+) -> Never = #externalMacro(module: "TestingMacros", type: "ExitTestBadCapturedValueMacro") where T: ~Copyable & ~Escapable
```

Comment:
Making sure I remember how this works correctly.. in this context, the `where T: ~Copyable & ~Escapable` means "`T` is allowed to be non-copyable or non-escapable", right?

Reply:
Correct. Widest possible net.